### ISSUE TITLE
Три простые функции для таймлайна.

### ISSUE THREAD

**Issue by beeheemooth** [morevnaproject/morevna-club#24]
(Это на "после Нового года")

Агний Игнатьев предложил сделать функциональные кнопки и (или) хоткеи для работы с кадрами на таймлайне по аналогии Flash. Речь идёт о клавишах F5, F6, F7.
Во Flash-е:

F5 - добавляет промежуточные (не ключевые) кадры вслед за выделенным на таймлайне, то есть продлевает время экспозиции ключевого кадра.

Shift + F5 - обратное действие: удаляет промежуточные кадры вслед выделенного, сжимает время.

F6 - копирует пердыдущий ключевой кадр и вставляет его в место где находится ползунок времени.

Shift + F6 - удаляет выделенный ключевой кадр. 

F7 - делает пустой ключевой кадр в месте где находится ползунок времени. (удаляется тем же Shift + F6)

В OpenToonz аналог F5 это простое растяжение, когда выделяется какой-нибудь кадр и тянется за ушко. А вот сжимать (как Shift+F5) не очень удобно, так как нужно сначала нужно выделить кадры и уже потом тянуть вверх (Xsheet). Также можно использовать Ctrl + C, Ctrl + V для добавления кадров, и Ctrl + X - для удаления.

Аналог F7 тоже есть: при установленной настройке Autocreation > Use Xsheet as Animation Sheet можно рисовать в любом месте уровня, новый ключевой кадр появляется автоматически (и в Xsheet и в Level Strip) как только стилус касается холста.

Но аналога F6 в OT нет. Скопировать ключевой кадр из одного места уровня в другое - это целая история: нужно сделать дубликат в Level Strip, затем с зажатым Alt-ом потянуть на Xsheet (таймлайн), далее нас ожидает другой сюрприз - кадр встал на своё место, но все последующие за ним кадры всё ещё упрямо принадлежат предыдущему. Их придётся удалить, оставив пустые ячейки, растянуть нужный нам кадр и благополучно отбуксовать съехавшую вниз последующую анимацию, вспоминая, где был следующий ключевой кадр.

Сделать это чуть проще, альтернативно, как здесь
![ezgif-3-b9219f0e26](https://user-images.githubusercontent.com/25724133/33772150-54e1631e-dc3c-11e7-934b-81ad397f6c32.gif)
в таймлайне не получится, _потому что кое-кто посчитал, что "не нужна нам такая вереница пустых слоев как в Xsheet" ^__^_

**Comment by Agnyy** [comment]
Спасибо Юра)) только думаю не хватает гифок как во флэше всё это выглядит. Ещё хорошо что F6 может разделить вытянутый кадр. НО самое главное, что ОТ очень хорошая программа, но у нас нету этих комбинаций, а эти комбинации ( я про манипуляции F5,F6,F7) вырабатывались годами во Flash и даёт огромное преимущества перед нами, но если мы себе это сделаем то мы полностью догоним Flash в по кадровой анимации, а то я буже услышал раз 40 вопросы про эти комбинации в ОТ, особенно от тех кто пришёл из Флэша им безумно этого не хватает. Да и сам я много раз сталкивался с необходимостью их. Всё суть что во Flash на этих комбинация построено 70% работы)) а у нас их нет, и даже аналогов их нет

**Comment by Agnyy** [comment]
Если мы сделаем эти команды, то мы сможем делать контейнерную анимацию, это очень, ну прям очень может ускорить работу))

**Comment by Agnyy** [comment]
А и ещё 
"в таймлайне не получится, потому что кое-кто посчитал, что "не нужна нам такая вереница пустых слоев как в Xsheet" ^__^" 
Это был я))) я посчитал что пустые столбци будут лишними))

**Comment by beeheemooth** [comment]
Раз надо гифы - вот:

**F5** и **Shift + F5**
![ezgif-3-ebe776d8ca](https://user-images.githubusercontent.com/25724133/33789746-9f4b56d4-dc83-11e7-9830-e1704b8cdd63.gif)

**F6** и **Shift + F6**
![ezgif-3-3a89f28470](https://user-images.githubusercontent.com/25724133/33789766-bc40f870-dc83-11e7-9aea-cef3b906c0d3.gif)

то же **F6** (копии как из пулемёта можно строчить)
![ezgif-3-7c50ef7ea1](https://user-images.githubusercontent.com/25724133/33789833-18916bdc-dc84-11e7-8aee-b5879ec4bb79.gif)
)

**F7** и **Shift + F6** (не Shift + F7 - это совсем другое)
![ezgif-3-c59c005cff](https://user-images.githubusercontent.com/25724133/33789890-6f394798-dc84-11e7-9c82-d4a5e189e9bd.gif)

**Comment by beeheemooth** [comment]
> Это был я))) я посчитал что пустые столбци будут лишними))

(っ°Д °; )っ   Да я во флэше делал кучу пустых слоев, чтобы можно было копировать, сдвигать, объединять, временно изолировать кадры, а в Xsheet-е это по умолчанию сетка, как футбольное поле, тут можно хоть жонглировать кадрами и уровнями.

**Comment by morevnaproject** [comment]
Да,, я помню как это во флеше и согласен, что коминации очень удобные и народ их любит. ^__^
В любом случае, спасибо за гифки - мне будет проще показать это Ивану.
